### PR TITLE
[Android/iOS] Fix programmatically open right SwipeItems

### LINKED ISSUE #10563: [Bug] SwipeView Open methods does not work for RightItems
Open method for SwipeView doesn't work for RightItems, it works for left items but not for right. Tested on Android and iOS.

### Steps to Reproduce

1. Add SwipeView to ViewPage
2. Add RightItems to that SwipeView
3. Invoke Open(OpenSwipeItem.RightItems) method on that SwipeView

### Expected Behavior
Open method should open view and reveal right items.

### Actual Behavior
Open method does nothing

### Basic Information
- Xamarin Forms: 4.6.0.726
- IDE: Visual Studio for Mac, ver 8.5.4 (build 12)
- Xcode ver 11.4.1

### Reproduction Link
https://github.com/mariaszek9003/SwipeViewTest
or you can either test it on your own SwipeView demo repository, just use right items instead of left on SwipeViewProgrammaticPage: https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/SwipeViewDemos

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Changes:
- Fix programmatically open right SwipeItems.
- Added animated parameter to open and close the SwipeView.

### Issues Resolved ### 

- fixes #10563

### API Changes ###
 
Added **Animated** parameter to `OpenSwipeEventArgs` and `CloseSwipeEventArgs`.
Now, can programmatically open and close the swipeView, animated or not.

```
void Open(OpenSwipeItem openSwipeItem, bool animated = true)
void Close(bool animated = true)
```

### Platforms Affected ### 

- iOS
- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

fix10563
animated-openclose-swipeview

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/SwipeEventArgs.cs [hunk 1]
```diff
@@ -12,14 +12,26 @@ protected BaseSwipeEventArgs(SwipeDirection swipeDirection)
 		public SwipeDirection SwipeDirection { get; set; }
 	}
 
+	public class CloseSwipeEventArgs : EventArgs
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply Xamarin.Forms.Core/SwipeEventArgs.cs]: Renamed.

**Comment by jsuarezruiz** [comment]
@PureWeen Added UITests.

File: Xamarin.Forms.Core/SwipeEventArgs.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hartez) [reply Xamarin.Forms.Core/SwipeEventArgs.cs]: Now that I look at this again, we should match the event names with the event arguments names as closely as we can. So now I think it should be `CloseRequestedEventArgs` and `OpenRequestedEventArgs`.

**Comment by PureWeen** [comment]
works great

Should the open behavior be blocked if it's already open in a different direction?

If I click Open Right and then click Open Left should it open Left now? Is there any reason to block that request if it's already open?

**Comment by hartez** [comment]
> Should the open behavior be blocked if it's already open in a different direction?
> 
> If I call Open Right and then call Open Left should it open Left now? Is there any reason to block that request if it's already open a different direction?

I can't think of any reason we should block it, as long as we can gracefully transition from one open state to the other.

**Comment by jsuarezruiz** [comment]
@hartez Renamed the SwipeView Open/Close EventArgs to `CloseRequestedEventArgs` and `OpenRequestedEventArgs`.

**Comment by jsuarezruiz** [comment]
Applied changes to no block the open behavior if it's already open in a different direction:
fix10563

**Review by PureWeen — CHANGES_REQUESTED** [review]
UI Tests are failing

**Comment by rmarinho** [comment]
@jsuarezruiz  tests are failing .. can you review please?

**Comment by jsuarezruiz** [comment]
@rmarinho All the tests are passing now.

**Review by hartez — APPROVED** [review]

**Comment by rmarinho** [comment]
@jsuarezruiz  seems its failing on iOS still..

**Comment by jsuarezruiz** [comment]
Captura de pantalla 2020-12-17 a las 18 42 55

**Review by rmarinho — APPROVED** [review]

**Pull request merged**